### PR TITLE
 Wireguard Docs Update

### PR DESCRIPTION
**Pull request by usman-umer**
Moved wireguard from the interfaces to vpn directory, as it is a VPN.
 Modified the index files to reflect this.
 Added Diagram of the VPN Topology
 Modified the weird IP subnets for local& remote sites. This Included:

	Changing the Interface Address for the wg01 tunnel
	Chaning the WAN addresses to addrs not in the RFC1918 range (1.1.1.1 & 2.2.2.2)
	Modifing the allowed IPs and static route to reflect this.
	Modified the ping test to reflect this.

 Added an annotated verison of the wireguard local&remote config.
 Added an example of a firewall exception for wirguard (OUTSIDE_IN)
 Modified the explanation for the behavior of AllowedIPs for remote clients.
 Added an example of "sudo wg" to the operational commands.
 Fixed typo

### CODE REVIEW

File: docs/configuration/vpn/wireguard.rst [hunk 1]
```diff
@@ -222,6 +287,24 @@ Operational Commands
 Status
 ======
 
+.. opcmd:: sudo wg
```
Comment (rebortg) [inline docs/configuration/vpn/wireguard.rst:290]: @c-po what do you think about this? It is not really an operation command. I have doubts to official mention sudo commands.

File: docs/configuration/interfaces/index.rst [hunk 2]
```diff
@@ -21,7 +21,6 @@ Interfaces
    tunnel
    vti
    vxlan
-   wireguard
```
Comment (rebortg) [inline docs/configuration/interfaces/index.rst:24]: i understand your point, but in the cli wireguard is under interfaces. So at the moment its must belong here.
See also a text about this [here](https://docs.vyos.io/en/latest/documentation.html#configuration-mode-pages)

File: docs/configuration/vpn/wireguard.rst [hunk 3]
```diff
@@ -10,10 +8,24 @@ WireGuard is an extremely simple yet fast and modern VPN that utilizes
 state-of-the-art cryptography. See https://www.wireguard.com for more
 information.
 
+*************
```
Comment (rebortg) [inline docs/configuration/vpn/wireguard.rst:11]: the headline over and/or underlines must be the same length as the headline. New sphinx versions will ignore this, but in read the docs will raise build warnings about this.

File: docs/configuration/vpn/wireguard.rst [hunk 4]
```diff
@@ -71,26 +84,39 @@ you want to tunnel (allowed-ips) to configure a WireGuard tunnel. The
...
 .. code-block:: none
 
-  set interfaces wireguard wg01 address '10.1.0.1/24'
+  set interfaces wireguard wg01 address '10.1.0.1/30'
   set interfaces wireguard wg01 description 'VPN-to-wg02'
-  set interfaces wireguard wg01 peer to-wg02 allowed-ips '10.2.0.0/24'
-  set interfaces wireguard wg01 peer to-wg02 address '192.168.0.142'
-  set interfaces wireguard wg01 peer to-wg02 port '12345'
+  set interfaces wireguard wg01 peer to-wg02 allowed-ips '192.168.2.0/24'
+  set interfaces wireguard wg01 peer to-wg02 address '2.2.2.2'
```
Comment (rebortg) [inline docs/configuration/vpn/wireguard.rst:94]: please don't use public IP space in examples, when it is not necessary.
reference also [here](https://docs.vyos.io/en/latest/documentation.html#address-space)

File: docs/configuration/vpn/wireguard.rst [hunk 5]
```diff
@@ -71,26 +84,39 @@ you want to tunnel (allowed-ips) to configure a WireGuard tunnel. The
...
+  set protocols static route 192.168.2.0/24 interface wg01
+
+**local side - annotated commands**
+
+.. code-block:: none
+
+  set interfaces wireguard wg01 address '10.1.0.1/30'                    # Address of the wg01 tunnel interface.          
+  set interfaces wireguard wg01 description 'VPN-to-wg02'
+  set interfaces wireguard wg01 peer to-wg02 allowed-ips '192.168.2.0/24'# Subnets that are allowed to travel over the tunnel
+  set interfaces wireguard wg01 peer to-wg02 address '2.2.2.2'           # Public IP of the peer
```
Comment (rebortg) [inline docs/configuration/vpn/wireguard.rst:107]: don't use public IP space, see above

File: docs/configuration/vpn/wireguard.rst [hunk 6]
```diff
@@ -106,33 +132,70 @@ or allows the traffic.
...
 .. code-block:: none
 
-  set interfaces wireguard wg01 address '10.2.0.1/24'
+  set interfaces wireguard wg01 address '10.1.0.2/30'
   set interfaces wireguard wg01 description 'VPN-to-wg01'
-  set interfaces wireguard wg01 peer to-wg02 allowed-ips '10.1.0.0/24'
-  set interfaces wireguard wg01 peer to-wg02 address '192.168.0.124'
-  set interfaces wireguard wg01 peer to-wg02 port '12345'
+  set interfaces wireguard wg01 peer to-wg02 allowed-ips '192.168.1.0/24'
+  set interfaces wireguard wg01 peer to-wg02 address '1.1.1.1'
```
Comment (rebortg) [inline docs/configuration/vpn/wireguard.rst:142]: don't use public IP space, see above

File: docs/configuration/vpn/wireguard.rst [hunk 7]
```diff
@@ -106,33 +132,70 @@ or allows the traffic.
...
+  set protocols static route 192.168.1.0/24 interface wg01
+
+**remote side - annotated commands**
+
+.. code-block:: none
+
+  set interfaces wireguard wg01 address '10.1.0.2/30'                    # Address of the wg01 tunnel interface.
+  set interfaces wireguard wg01 description 'VPN-to-wg01'
+  set interfaces wireguard wg01 peer to-wg02 allowed-ips '192.168.1.0/24'# Subnets that are allowed to travel over the tunnel
+  set interfaces wireguard wg01 peer to-wg02 address '1.1.1.1'           # Public IP address of the Peer
```
Comment (rebortg) [inline docs/configuration/vpn/wireguard.rst:155]: don't use public IP space, see above

**Review by rebortg — CHANGES_REQUESTED** [review]
Hey, thanks for your work. i added some feedback and request some changes

File: docs/configuration/interfaces/index.rst (hunk 2 above)
Comment (usman-umer) [reply docs/configuration/interfaces/index.rst:24]: Sure

File: docs/configuration/vpn/wireguard.rst (hunk 4 above)
Comment (usman-umer) [reply docs/configuration/vpn/wireguard.rst:94]: Sure, though I will make it clear that it is a public IP address. Something like <WAN Addr>
When I tried to configure this myself, it was unclear what IP address was the tunnel and what the WAN address was.

File: docs/configuration/vpn/wireguard.rst (hunk 1 above)
Comment (usman-umer) [reply docs/configuration/vpn/wireguard.rst:290]: Regardless of its position in the doc, I think it should still be present. It provides diagnostics such as the latest handshake and traffic rx & tx, that isn't available through any of the other commands.

**Comment by usman-umer** [comment]
> Hey, thanks for your work. i added some feedback and request some changes

You're welcome. Resolved some issues and made some amendments.

File: docs/configuration/vpn/wireguard.rst (hunk 1 above)
Comment (c-po) [reply docs/configuration/vpn/wireguard.rst:290]: We do not document such raw commands. If the diagnostic data is usefule, we should rather add a new `show interfaces wireguard wg0 detail` command

Comment (usman-umer) [reply docs/configuration/vpn/wireguard.rst:290]: Sure, will commit later.

**Comment by usman-umer** [comment]
Just pushed my changes, let me know if you have any questions.

**Comment by rebortg** [comment]
To show that these are public IPs, you can use the addresses here:
https://docs.vyos.io/en/equuleus/documentation.html#address-space

tl;dr
There are space for documentation:
192.0.2.0/24
198.51.100.0/24
203.0.113.0/24
2001:db8::/32

but i will merge it for now, thank you very much

**Comment by usman-umer** [comment]
@rebortg You're welcome, but just noticed that you only merged my branch into the master branch. Shouldn't it be merged into the other crux & equuleus branches, as the changes are still applicable to the stable versions?
